macros/assertions_filter.sql

```diff
@@ -64,3 +64,29 @@ EXISTS (
 {%- endif -%}
 
 {%- endmacro %}
+
+
+{%- macro snowflake__assertions_filter(column, exclude_list, include_list, reverse) -%}
+
+{#- Check if both exclude_list and include_list are provided -#}
+{%- if exclude_list is not none and include_list is not none -%}
+    {{
+        exceptions.raise_compiler_error(
+            'exclude_list or include_list must be provided. Not both.'
+            ~ 'Got (exclude_list: ' ~ exclude_list 
+            ~ ', include_list: ' ~ include_list ~ ')'
+        )
+    }}
+{%- endif -%}
+
+{#- Generate filtering expression  -#}
+{{- 'NOT ' if reverse else '' -}}
+{%- if include_list is not none -%}
+ARRAY_SIZE(ARRAY_INTERSECTION({{ column }},ARRAY_CONSTRUCT('{{ include_list | join("\', \'")}}'))) = 0
+{%- elif exclude_list is not none -%}
+ARRAY_SIZE(ARRAY_EXCEPT({{ column }},ARRAY_CONSTRUCT('{{ exclude_list | join("\', \'")}}'))) = 0
```

Comment:
Great job with intersection and union !

Reply:
This should also work with databricks (and maybe other adapters as well) I think